core/src/main/java/io/github/lsposed/lspd/service/PackageService.java

```diff
@@ -176,4 +195,92 @@ private static PackageInfo getPackageInfoWithComponents(String packageName, int
             return null;
         return pkgInfo;
     }
+
+    static abstract class IntentSenderAdaptor extends IIntentSender.Stub {
+        public abstract void send(Intent intent);
+
+        @Override
+        public int send(int code, Intent intent, String resolvedType, IIntentReceiver finishedReceiver, String requiredPermission, Bundle options) {
+            send(intent);
+            return 0;
+        }
+
+        @Override
+        public void send(int code, Intent intent, String resolvedType, IBinder whitelistToken, IIntentReceiver finishedReceiver, String requiredPermission, Bundle options) {
+            send(intent);
+        }
+
+        public IntentSender getIntentSender() throws IllegalAccessException, InvocationTargetException, InstantiationException, NoSuchMethodException {
+            @SuppressWarnings("JavaReflectionMemberAccess")
+            Constructor<IntentSender> intentSenderConstructor = IntentSender.class.getConstructor(IIntentSender.class);
+            intentSenderConstructor.setAccessible(true);
+            return intentSenderConstructor.newInstance(this);
+        }
+    }
+
+    public static boolean uninstallPackage(VersionedPackage versionedPackage) throws RemoteException, InterruptedException, InvocationTargetException, NoSuchMethodException, InstantiationException, IllegalAccessException {
+        CountDownLatch latch = new CountDownLatch(1);
+        final boolean[] result = {false};
+        pm.getPackageInstaller().uninstallExistingPackage(versionedPackage, "com.android.shell", new IntentSenderAdaptor() {
+            @Override
+            public void send(Intent intent) {
+                int status = intent.getIntExtra(PackageInstaller.EXTRA_STATUS, PackageInstaller.STATUS_FAILURE);
+                result[0] = status == PackageInstaller.STATUS_SUCCESS;
+                latch.countDown();
+            }
+        }.getIntentSender(), 0);
+        latch.await();
+        return result[0];
```

Comment:
what's the point of returning a boolean if the return value is not used?

Reply:
It's for the manager.